mgc/terraform-provider-mgc/mgc/resources/mgc_dbaas_instances_snapshots.go

```diff
@@ -0,0 +1,224 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	mgcSdk "magalu.cloud/lib"
+	dbaasSnapshots "magalu.cloud/lib/products/dbaas/instances/snapshots"
+	"magalu.cloud/terraform-provider-mgc/mgc/client"
+	"magalu.cloud/terraform-provider-mgc/mgc/tfutil"
+)
+
+const snapshotStatusTimeout = 30 * time.Minute
```

Comment:
https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/retries-and-customizable-timeouts